### ISSUE TITLE
Support Neo4j 2.2 [DATAGRAPH-593]

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-data-neo4j#1155]
**[Jonathan Crosmer](https://jira.spring.io/secure/ViewProfile.jspa?name=jcrosmer)** opened **[DATAGRAPH-593](https://jira.spring.io/browse/DATAGRAPH-593?redirect=false)** and commented

Currently Spring Data Neo4j depends on Neo4j 2.1.7, and seems incompatible with 2.2.  I ran into problems when I was running 2.2 server and connecting with Spring Data 3.3 using SpringCypherRestGraphDatabase.  The errors were pretty strange (e.g., SDN would try to add a node, then add to types index, but adding to index would fail with "node X not found" NotFoundException for the node that was just created), so I assumed it was just some version incompatibility.
Neo4j 2.2 promises massive performance improvements, so this would be especially nice to have

---

**Affects:** 3.3 GA (Fowler)

**Backported to:** [3.3.1 (Fowler SR1)](https://github.com/spring-projects/spring-data-neo4j/milestone/61?closed=1)

3 votes, 4 watchers
